kratos/utilities/function_parser_utility.cpp

```diff
@@ -61,19 +103,7 @@ BasicGenericFunctionUtility::BasicGenericFunctionUtility(const std::string& rFun
     InitializeParser();
 
     // Check if it depends on space
-#if !(defined(__GNUC__) && !defined(__clang__) && (__GNUC__ < 4 || (__GNUC__ == 4 && (__GNUC_MINOR__ < 9))))
```

Comment:
My question is if it is more efficient keep this check and do the new method only in older compilers

Reply:
Well the issue is that splitting the code is what led to these bugs being uncaught.

Besides, std::regex is notoriously slow, so I wouldn't be surprised if it was slower than the hand-rolled implementation. But no way of knowing this without testing.

However, since this piece of code is only ran a few times before starting the simulation, performance is not critical.